File: spec/cypress/integration/essentials.spec.js
describe('Essentials', () => {
  it('has tiles with values', () => {
    cy.visit('/essentials');

    cy.get('#tile-total_inverter_power-now').should('contain', '10,0');
    cy.get('#tile-total_inverter_power-day').should('contain', '20,0');
    cy.get('#tile-total_inverter_power-month').should('contain', '20,0');
    cy.get('#tile-total_inverter_power-year').should('contain', '20,0');
    cy.get('#tile-co2_reduction-year a').should('contain', '8');
    cy.get('#tile-savings-year a').should('contain', '3,68');
  });
});

File: spec/system/essentials_spec.rb
describe 'Essentials' do
  include ActiveSupport::Testing::TimeHelpers

  before { travel_to Time.zone.local(2022, 6, 21, 12, 0, 0) }

  it 'displays tiles with values' do
    visit '/essentials'

    expect(page).to have_css('#tile-inverter_power-now')
    expect(page).to have_css('#tile-inverter_power-day')
    expect(page).to have_css('#tile-inverter_power-month')
    expect(page).to have_css('#tile-inverter_power-year')
    expect(page).to have_css('#tile-co2_reduction-year')
    expect(page).to have_css('#tile-savings-year')

    # Check for specific values (the exact values might vary based on test data)
    within '#tile-inverter_power-now' do
      expect(page).to have_content(/\d+([.,]\d+)?/)
    end

    within '#tile-co2_reduction-year' do
      expect(page).to have_content(/\d+/)
    end

    within '#tile-savings-year' do
      expect(page).to have_content(/\d+/)
    end
  end
end
